app/controllers/RequestLock.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import play.modules.reactivemongo.ReactiveMongoComponent
+import uk.gov.hmrc.lock.{LockKeeper, LockMongoRepository, LockRepository}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RequestLock @Inject() (reactiveMongoComponent: ReactiveMongoComponent)(
+    implicit
+    appConfig: AppConfig, ec: ExecutionContext) {
+
+  def tryLock[T](lockName: String)(body: => Future[T])(implicit ec: ExecutionContext): Future[Option[T]] = {
+    lazy val lockKeeper = new LockKeeper {
```

Comment:
Maybe this could be part of a class instead of being computed each time the function is called?

Reply:
My thinking was that the tryLock was a utility, not part of the controller and this pulls in mongo, not something that's normally associated with the controller